### PR TITLE
Initial attempt at spec downloading

### PR DESCRIPTION
**Pull request by j-f1**
I started this in, like, September but never finished it. Current status:

```
Error: The following types are undefined:
	- LinkStyle
	- MediaSource
	- MediaStream
	- MouseEvent
	- MouseEventInit
	- SVGImageElement
	- SVGScriptElement
	- VisibilityState
	- WebGL2RenderingContext
	- WindowProxy
```

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
I wonder if WebGL 1 and WebGL 2 IDL is overlapping. If not, we could apply codegen on both 🤔

**Comment by j-f1** [comment]
I tried pulling in webgl 2 and it seems to depend on webgl 1. So it should be ok to pull it down.

File: deps/dl-specs.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+bs-to-idl/from-github.sh https://github.com/w3c/webappsec-referrer-policy/blob/master/index.src.html webidl-sources/referrer-policy.webidl &
+bs-to-idl/from-github.sh https://github.com/heycam/webidl/blob/master/index.bs webidl-sources/webidl.webidl noindent &
+bs-to-idl/from-github.sh https://github.com/w3c/FileAPI/blob/master/index.bs webidl-sources/file.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/hr-time/blob/gh-pages/index.html webidl-sources/hr-time.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/ServiceWorker/blob/master/docs/index.bs webidl-sources/service-worker.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/fxtf-drafts/blob/master/geometry/Overview.bs webidl-sources/geometry.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/aria/blob/master/index.html webidl-sources/aria.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/aria/blob/master/index.html webidl-sources/aria.webidl &
+bs-to-idl/from-github.sh https://github.com/w3c/csswg-drafts/blob/master/cssom-1/Overview.bs webidl-sources/cssom.webidl &
+curl https://www.khronos.org/registry/webgl/specs/1.0.2/webgl.idl > webidl-sources/webgl.webidl
```
Comment (MaxDesiatov) [inline deps/dl-specs.sh:29]:
```suggestion
curl https://www.khronos.org/registry/webgl/specs/1.0.2/webgl.idl > webidl-sources/webgl.webidl
curl https://www.khronos.org/registry/webgl/specs/2.0/webgl2.idl > webidl-sources/webgl2.webidl
```

**Comment by j-f1** [comment]
Looks like the WebGL 1 spec uses a `typedef` inside an interface, which our parser doesn’t support

**Comment by MaxDesiatov** [comment]
Thanks for investigation! I saw that error, but wasn't sure where it's coming from. Would you have time for fixing the parser soon, or would you like me to tackle it?

**Comment by j-f1** [comment]
If you want to you can but I’m pretty much free the whole week

**Comment by MaxDesiatov** [comment]
I got some progress with the parser fix here in the meantime https://github.com/Apodini/webidl2swift/pull/12

File: deps/dl-specs.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# TODO: https://github.com/w3c/uievents (multiple files)
+# TODO: https://svgwg.org/svg2-draft/ (multiple files)
+# TODO: https://github.com/w3c/page-visibility/blob/gh-pages/index.html
+# TODO: something special for WindowProxy (maybe just typealias it to Window? because the actual Window is inaccessible)
+
+wait
+
+
+# Not currently used:
+# bs-to-idl/from-github.sh https://github.com/whatwg/xhr/blob/master/xhr.bs webidl-sources/xhr.webidl &
```
Comment (MaxDesiatov) [inline deps/dl-specs.sh:43]: This `.bs` file is already mentioned above

```suggestion
```

**Comment by MaxDesiatov** [comment]
Looks like WebGL spec parsing isn't fixed with that PR, I'm adding proper diagnostics to `webidl2swift` now anyway to investigate this properly...

**Comment by j-f1** [comment]
I think this is the wrong approach. Instead, I’m going to try pulling the `webidl2swift` code into this repository, and taking advantage of the W3C’s official (JS-based) parser instead of using the hand-rolled Swift one.

**Pull request closed without merging**